Engine/DataFeeds/LiveSynchronizer.cs

```diff
@@ -30,6 +30,11 @@ public class LiveSynchronizer : Synchronizer
     {
         private readonly AutoResetEvent _newLiveDataEmitted = new AutoResetEvent(false);
 
+        /// <summary>
+        /// Maximum time to wait for new live data before synchronizing the data feed subscriptions
+        /// </summary>
+        protected virtual TimeSpan NewLiveDataTimeout { get; } = TimeSpan.FromMilliseconds(500);
```

Comment:
Do we always create a sub class for the unit tests? If so, then disregard this suggestion, but thinking it would be easier to just make it a public get/set.

Reply:
We already have a subclass in the `LiveTradingDataFeed` tests (`TestableLiveSynchronizer`),
which I was thinking to change to something like this:

``` C#
    internal class TestableLiveSynchronizer : LiveSynchronizer
    {
        private readonly ITimeProvider _timeProvider;

        protected override TimeSpan NewLiveDataTimeout { get; }

        public TestableLiveSynchronizer(ITimeProvider timeProvider = null, TimeSpan? newLiveDataTimeout = null)
        {
            _timeProvider = timeProvider ?? new RealTimeProvider();
            NewLiveDataTimeout = newLiveDataTimeout ?? base.NewLiveDataTimeout;
        }

        protected override ITimeProvider GetTimeProvider()
        {
            return _timeProvider;
        }
    }
```
Then, only in specific tests:
``` C#
var synchronizer = new TestableLiveSynchronizer(timeProvider, TimeSpan.FromMilliseconds(50));
```

Reply:
Awesome sauce